kernel/data-pipeline/scenario/cdc/core/src/main/java/org/apache/shardingsphere/data/pipeline/cdc/api/CDCJobAPI.java

```diff
@@ -257,10 +257,14 @@ public void disable(final String jobId) {
      * Drop CDC job.
      *
      * @param jobId job id
+     * @param force force
      */
-    public void drop(final String jobId) {
+    public void drop(final String jobId, final boolean force) {
         CDCJobConfiguration jobConfig = jobConfigManager.getJobConfiguration(jobId);
-        ShardingSpherePreconditions.checkState(PipelineJobIdUtils.getElasticJobConfigurationPOJO(jobId).isDisabled(), () -> new PipelineInternalException("Can't drop streaming job which is active"));
+        if (!force) {
+            ShardingSpherePreconditions.checkState(PipelineJobIdUtils.getElasticJobConfigurationPOJO(jobId).isDisabled(),
+                    () -> new PipelineInternalException("Can't drop streaming job which is active"));
+        }
```

Comment:
Could we just remove `checkState` to allow drop when job is active?

I just planed to remove it, since CDC job could not be dropped when CDC E2E test failure in some cases.

Reply:
I move `checkState` method, make sure DistSQL could drop job anytime.